master/buildbot/statistics/capture.py

```diff
@@ -80,25 +80,74 @@ def consume(self, routingKey, msg):
         send them to the storage backends.
         """
         builder_info = yield self.master.data.get(("builders", msg['builderid']))
-        if self._builder_name == builder_info['name']:
+
+        if self._builder_name_matches(builder_info):
             properties = yield self.master.data.get(("builds", msg['buildid'], "properties"))
-            try:
-                ret_val = self._callback(properties, self._property_name)
-            except KeyError:
-                config.error("CaptureProperty failed."
-                             " The property %s not found for build number %s on builder %s."
-                             % (self._property_name, msg['number'], self._builder_name))
-            context = self._defaultContext(msg)
-            series_name = '%s-%s' % (self._builder_name, self._property_name)
-            post_data = {
-                "name": self._property_name,
-                "value": ret_val
-            }
-            yield self._store(post_data, series_name, context)
+            filtered_prop_names = []
+            if self._regex:
+                prop_names = properties.keys()
+                for pn in prop_names:
```

Comment:
If you want prop_names as a list of keys, please use ```list(properties)``` or (less preferably) ```list(properties.keys())```.
If you only want to use it for iteration, please use ```for pn in properties:```. 

This is to ensure python3 compatibility